src/HotChocolate/Filters/src/Types.Filters/Conventions/Filter/FilterConventionDescriptor.cs

```diff
@@ -0,0 +1,171 @@
+                kind,
+                (FilterKind kind) => FilterConventionTypeDescriptor.New(this, kind))
+                .Ignore(ignore);
+            return this;
+        }
+
+        public IFilterConventionDescriptor AddImplicitFilter(
+            TryCreateImplicitFilter factory,
+            int? position = null)
+        {
+            if (factory == null)
+            {
+                throw new ArgumentNullException(nameof(factory));
+            }
+
+            var insertAt = position ?? _implicitFilters.Count;
+            _implicitFilters.Insert(insertAt, factory);
+
+            return this;
+        }
+
+        public FilterConventionDefinition CreateDefinition()
+        {
+            if (_visitorDescriptor == null)
+            {
+                throw new SchemaException(
+                    SchemaErrorBuilder.New()
+                    .SetMessage("No visitor was defined for FilterConvention")
+                    .Build());
+            }
+
+            var allowedOperations
+                = new Dictionary<FilterKind, IReadOnlyCollection<FilterOperationKind>>();
+            var typeDefinitions = new Dictionary<FilterKind, FilterConventionTypeDefinition>();
+            var defaultOperationNames = new Dictionary<FilterOperationKind, CreateFieldName>();
+            var defaultOperationDescriptions = new Dictionary<FilterOperationKind, string>();
+
+            foreach (FilterConventionTypeDescriptor descriptor in _configurations.Values)
+            {
+                FilterConventionTypeDefinition definition = descriptor.CreateDefinition();
+                if (!definition.Ignore)
+                {
+                    typeDefinitions[definition.FilterKind] = definition;
+                    allowedOperations[definition.FilterKind] = definition.AllowedOperations;
+                }
+            }
+
+            foreach (FilterConventionDefaultOperationDescriptor descriptor
+                in _defaultOperations.Values)
+            {
+                FilterConventionOperationDefinition definition = descriptor.CreateDefinition();
+
+                if (!definition.Ignore)
+                {
+                    if (definition.Description != null)
+                    {
+                        defaultOperationDescriptions[definition.OperationKind]
+                            = definition.Description;
+                    }
+
+                    if (definition.Name != null)
+                    {
+                        defaultOperationNames[definition.OperationKind]
+                            = definition.Name;
+                    }
+                }
+            }
+            Definition.DefaultOperationDescriptions = defaultOperationDescriptions;
+            Definition.DefaultOperationNames = defaultOperationNames;
+            Definition.AllowedOperations = allowedOperations;
+            Definition.TypeDefinitions = typeDefinitions;
+            Definition.ImplicitFilters = _implicitFilters.ToArray();
```

Comment:
why ToArray?

Reply:
To follow the pattern of immutability. we should look at this this week